src/orion/core/utils/tests.py

```diff
@@ -0,0 +1,180 @@
+        if from_yaml is not None:
+            with open(from_yaml) as f:
+                exp_config = list(yaml.safe_load_all(f))
+                experiments = exp_config[0]
+                trials = exp_config[1]
+
+        self.experiments = _select(experiments, [])
+        self.trials = _select(trials, [])
+        self.workers = _select(workers, [])
+        self.resources = _select(resources, [])
+
+    def init(self):
+        """Initialize environment before testing"""
+        self.storage()
+        self.database = get_storage()._db
+        self.load_experience_configuration()
+        return self
+
+    def get_experiment(self, name, user=None, version=None):
+        """Make experiment id deterministic"""
+        exp = Experiment(name, user=user, version=version)
+        exp._id = name
+        return exp
+
+    def get_trial(self, index):
+        """Return a Trial"""
+        return Trial(**self.trials[index])
+
+    def cleanup(self):
+        """Cleanup after testing"""
+        _remove(self.tempfile)
+
+    def load_experience_configuration(self):
+        """Load an example database."""
+        for i, t_dict in enumerate(self.trials):
+            self.trials[i] = Trial(**t_dict).to_dict()
+
+        # sort by submit time like the protocol does
+        self.trials.sort(key=lambda obj: int(obj['_id'], 16), reverse=True)
+
+        for i, _ in enumerate(self.experiments):
+            path = os.path.join(
+                os.path.dirname(__file__),
+                self.experiments[i]["metadata"]["user_script"])
+
+            self.experiments[i]["metadata"]["user_script"] = path
+            self.experiments[i]['version'] = 1
+            self.experiments[i]['_id'] = i
+
+        self.database.write('experiments', self.experiments)
+        self.database.write('trials', self.trials)
+        self.database.write('workers', self.workers)
+        self.database.write('resources', self.resources)
+
+    def __enter__(self):
+        """Load a new database state"""
+        _, self.tempfile = tempfile.mkstemp('orion_test')
+
+        for singleton in self.SINGLETONS:
+            self.new_singleton(singleton, new_value=None)
+
+        self.cleanup()
+        return self.init()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Cleanup database state"""
+        self.cleanup()
+
+        for obj in self.singletons:
+            self.restore_singleton(obj)
+
+    def new_singleton(self, obj, new_value=None):
+        """Replace a singleton by another value"""
+        self.singletons[obj] = obj.instance
+        obj.instance = new_value
+
+    def restore_singleton(self, obj):
+        """Restore a singleton to its previous value"""
+        obj.instance = self.singletons.get(obj)
+
+    def storage(self, storage_type='legacy'):
+        """Return test storage"""
+        try:
+            config = {
+                'database': {
+                    'type': 'PickledDB',
```

Comment:
All tests running with OrionState are on PickledDB but the rest not using OrionState is still on MongoDB right?

Reply:
yes